### PR TITLE
Update init behavior

### LINKED ISSUE #9: Update `init` behavior
When using `useReducer`, init returns the state:

```js
useReducer(reducer, initialState, () => {
  return {
    someState: true
  };
})
```

With the current API of this library, you must return the "internal" structure of the store:

```js
useReducerWithSideEffects(reducer, initialState, () => {
  return {
    state: { someState: true },
    sideEffects: [] // you also must include this or else an uncaught exception will occur
  };
})
```

Users are unable to use `Update`/`UpdateWithSideEffects`/etc., because those don't return the correct key: they return `newState` rather than `state`.

I suggest refactoring `Update`/etc. to internally use the key `state` instead. That way, users of `init` can use a consistent API with this lib:

```js
useReducerWithSideEffects(reducer, initialState, () => {
  return Update({ someState: true });
})
```

This change could even be released as a bug fix, I think, since it only changes non-public structures.

I'm curious to hear your thoughts on this @conorhastings !

### PR DESCRIPTION
**Pull request by jamesplease**
This PR has two commits that to do two separate things:

de98e12
This updates #11 to fully fix #9 by taking `sideEffects` into account. It ensures that the sideEffects from `init` replace any sideEffects set up in `initialState`.

5ff7362
This is slightly tangential but it renames the `newSideEffect` argument to be `sideEffects`. This makes it so that the update shape mirrors the internal state.

Previously, the shapes were:

```js
// Update
{
  newState,
  newSideEffect
}

// State
{
  state,
  sideEffects
}
```

After this commit, the shapes are:

```js
// Update
{
  state,
  sideEffects
}

// State
{
  state,
  sideEffects
}
```

I think this naming change makes sense in light of #10 . Users should only need to think about having one state imo.

---

I tested and this seems to be working in my app where I'm using this lib.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -27,22 +27,29 @@ async function executeSideEffects({ sideEffects, state, dispatch }) {
...
+
+  const newSideEffects = newState.sideEffects
+      ? [
+          ...existingEffects,
+          ...(Array.isArray(newState.sideEffects) ? newState.sideEffects : [newState.sideEffects]),
+        ]
+      : state.sideEffects;
+
+  return {
+    state: newState.state || prevState.state,
```
Comment (jamesplease) [inline src/index.js:41]: We may not want to fall back to `prevState.state` this when this fn is called from `init`...what do you think @conorhastings ?

**Comment by conorhastings** [comment]
have interview during day, will review all tonight

**Comment by jamesplease** [comment]
No rush at all! Good luck today👌

File: src/index.js (hunk 1 above)
Comment (conorhastings) [reply src/index.js:41]: i think you're right, the side effects can return a valid null state or 0 state

Comment (conorhastings) [reply src/index.js:41]: i think if we just change this, tthis and compose are good to go, tried itt out in sample app

Comment (conorhastings) [reply src/index.js:41]: thanks for making this so much better lol

**Comment by conorhastings** [comment]
closed by #19

**Pull request closed without merging**